platform-sdk/consensus-otter-tests/src/testOtter/java/org/hiero/otter/test/StartFromStateTest.java

```
@@ -77,8 +77,18 @@ void migrationTest(final int numberOfNodes, @NonNull final TestEnvironment env)
                 .haveConsistentRounds();
         assertContinuouslyThat(network.newReconnectResults()).doNotAttemptToReconnect();
 
+        final AtomicBoolean upgradeDetectedLogMsgFound = network.nodes()
+                .getFirst()
+                .newLogResult()
+                .onNextMatch(logEntry -> logEntry.message().contains(OtterApp.UPGRADE_DETECTED_LOG_PAYLOAD));
+
         network.start();
 
+        timeManager.waitForCondition(
+                upgradeDetectedLogMsgFound::get,
+                Duration.ofMinutes(2),
+                "Could not find the log message indicating a upgraded happened");
```

Is it really necessary to wait for this message? The test would be simpler if we did this check at the end.

---

Changed as you suggested. Initial thinking was to fail fast, but verifying at the end is simpler.